### PR TITLE
Add formats tags

### PR DESCRIPTION
**Pull request by lhoestq**
Added `formats: list[str]` to "hub-cache". It is computed in "compatible-libraries" since compatibilities depend on the format.

Possibles values are
- csv
- json
- parquet
- text
- imagefolder
- audiofolder

Mentioned in https://github.com/huggingface/datasets-server/issues/2455

I chose to have it as a list for simplicity, but lmk if you think we should set it to `format: Optional[str]` instead

includes a minor change: updated the libraries order

### CODE REVIEW

File: services/worker/src/worker/dtos.py [hunk 1]
```diff
@@ -221,6 +221,7 @@ class IsValidResponse(TypedDict):
 
 DatasetTag = Literal["croissant"]
 DatasetLibrary = Literal["mlcroissant", "webdataset", "datasets", "pandas", "dask"]
+DatasetFormat = Literal["json", "csv", "parquet", "imagefolder", "audiofolder", "webdataset", "text"]
```
Comment (severo) [inline services/worker/src/worker/dtos.py:224]: we cannot discriminate between json and jsonl, right?

File: services/worker/src/worker/dtos.py [hunk 2]
```diff
@@ -240,6 +241,7 @@ class CompatibleLibrary(TypedDict):
 class DatasetCompatibleLibrariesResponse(TypedDict):
     tags: list[DatasetTag]
     libraries: list[CompatibleLibrary]
+    formats: list[DatasetFormat]
```
Comment (severo) [inline services/worker/src/worker/dtos.py:244]: having two formats means that two configs are in different formats, right?

File: services/worker/src/worker/job_runners/dataset/compatible_libraries.py [hunk 3]
```diff
@@ -607,20 +619,22 @@ def compute_compatible_libraries_response(
...
         if builder_name in get_compatible_library_for_builder:
             try:
                 compatible_library = get_compatible_library_for_builder[builder_name](dataset, hf_token)
                 libraries.append(compatible_library)
             except NotImplementedError:
                 pass
-    return DatasetCompatibleLibrariesResponse(tags=tags, libraries=libraries)
+        # mlcroissant library
+        libraries.append(get_mlcroissant_compatible_library(dataset, infos, partial=partial))
+        tags.append("croissant")
```
Comment (severo) [inline services/worker/src/worker/job_runners/dataset/compatible_libraries.py:636]: good idea to move them to the end of the libraries list (as we will respect this order on the Hub)

**Review by severo — APPROVED** [review]

File: services/worker/src/worker/dtos.py (hunk 2 above)
Comment (lhoestq) [reply services/worker/src/worker/dtos.py:244]: Currently having multiple formats is not supported (only one is detected)

File: services/worker/src/worker/dtos.py (hunk 1 above)
Comment (lhoestq) [reply services/worker/src/worker/dtos.py:224]: We could differentiate but it requires to read some data. For now I'd keep it under "json"

File: services/worker/src/worker/dtos.py (hunk 2 above)
Comment (severo) [reply services/worker/src/worker/dtos.py:244]: Ok. I was wondering why we return a list of formats

**Pull request merged**